File: python/56b_1547A_ShortestPathWithObstacle.py
def solve(xa, ya, xb, yb, xf, yf):
    if xa == xb and xf == xa and min(ya,yb) < yf < max(ya,yb):
        return abs(xa - xb) + abs(ya - yb) + 2
    elif ya == yb and yf == ya and min(xa,xb) < xf < max(xa,xb):
        return abs(xa - xb) + abs(ya - yb) + 2
    else:
        return abs(xa - xb) + abs(ya - yb) 

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        temp = input()
        xa, ya = input().split()
        xb, yb = input().split()
        xf, yf = input().split()
        ans.append(solve(int(xa), int(ya), int(xb), int(yb), int(xf), int(yf)))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/56b_1547A_ShortestPathWithObstacle.cpp
#include<bits/stdc++.h>

int solve(int xa, int ya, int xb, int yb, int xf, int yf){
    if(abs(xa - xb) == 0 && xf == xa && std::min(ya, yb) < yf && std::max(ya, yb) > yf){
        return abs(xa - xb) + abs(ya - yb) + 2;
    }
    else if (abs(ya - yb) == 0 && yf == ya && std::min(xa, xb) < xf && std::max(xa, xb) > xf){
        return abs(xa - xb) + abs(ya - yb) + 2;
    }
    else
        return abs(xa - xb) + abs(ya - yb);
}

int main(){
    int t, xa, xb, xf, ya, yb, yf;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>xa>>ya;
        std::cin>>xb>>yb;
        std::cin>>xf>>yf;
        ans.push_back(solve(xa, ya, xb, yb, xf, yf));
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}